Squash

Maickonn opened the pull request:
Added squash action.

Maickonn commented:
Hello, hackhaslam!

Can you review this pull request and tell me if you need to change anything to merge?

I'm waiting.

A big hug!

Maickonn commented:
@hackhaslam you have any position about this pull request?

hackhaslam commented:
I'm sorry. I haven't been able to review it yet. I will try to do it soon.

Maickonn commented:
Okay, thanks.

tommai78101 commented on src/ui/MenuBar.cpp:
```diff
@@ -481,6 +481,19 @@ MenuBar::MenuBar(QWidget *parent)
   });
 
+  mRebase = branch->addAction(tr("Squash..."));
+  mRebase->setShortcut(tr("Ctrl+Shift+S"));
```
Quick question - Is CTRL+SHIFT+S an universal hotkey shortcut for most git frontend applications for a Squash action?

Maickonn replied:
Hi! Sorry for the delay. I searched and found no shortcut key patterns for squash.

tommai78101 replied:
Very well, I think it's fair enough.

Maickonn replied:
Right, no problem.

hackhaslam replied:
You're resetting the `mRebase` member instead of setting the `mSquash` one you added.

Maickonn commented:
@hackhaslam if possible, review this pull request. We need this function. 😅

Maickonn commented:
Wow, sorry me!

I will fix it.

On Sat, Dec 28, 2019, 4:59 PM Jason Haslam <notifications@github.com> wrote:

> *@hackhaslam* commented on this pull request.
> ------------------------------
>
> In src/ui/MenuBar.cpp
> <https://github.com/gitahead/gitahead/pull/214#discussion_r361811250>:
>
> > @@ -481,6 +481,19 @@ MenuBar::MenuBar(QWidget *parent)
>      dialog->open();
>    });
>
> +  mRebase = branch->addAction(tr("Squash..."));
> +  mRebase->setShortcut(tr("Ctrl+Shift+S"));
>
> You're resetting the mRebase member instead of setting the mSquash one
> you added.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/gitahead/gitahead/pull/214?email_source=notifications&email_token=AAKNNM2ZG32UGRMIIBMC3D3Q26V3VA5CNFSM4HV2T2BKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCQKR7AA#discussion_r361811250>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAKNNM7EFZ36TFLMZ3MHWH3Q26V3VANCNFSM4HV2T2BA>
> .
>

hackhaslam commented:
@Maickonn Actually, maybe hold off a minute before making any changes. Is this meant to implement `merge --squash`? In that case, I don't know if it really merits its own top-level menu item.

hackhaslam commented:
I merged this mostly as is. I fixed the menu bar issue I mentioned before, and changed the shortcut to Ctrl+Shift+Q since Ctrl+Shift+S is the typical shortcut for 'Save As'.

Maickonn commented:
> I merged this mostly as is. I fixed the menu bar issue I mentioned before, and changed the shortcut to Ctrl+Shift+Q since Ctrl+Shift+S is the typical shortcut for 'Save As'.

It's all right. I'm sorry for not fixing the commit problem as you informed me in the review. I have been distracted for a while, but I will participate more actively.

Thank you for having accepted and corrected the pull request.

hackhaslam commented:
Nah, it's alright. I didn't really expect you to fix that since I told you to hold off. I'm sorry it took me so long to circle back around to this.

0xCAP commented:
When is this feature gonna be included in a release? Latest update still didn't have it.

Maickonn commented:
> When is this feature gonna be included in a release? Latest update still didn't have it.

Hello!

This is are available on latest update. Make right click on any branch, it will show below to the merge button.
